internal/network/ports.go

```diff
@@ -200,3 +193,37 @@ func getDefaultPort(outputType obs.OutputType, urlStr string) int32 {
 	}
 	panic(fmt.Sprintf("unknown output type: %s", outputType))
 }
+
+// GetProxyPorts extracts unique ports from cluster-wide proxy environment variables.
+// It parses HTTP_PROXY and HTTPS_PROXY URLs to determine explicit proxy ports,
+// or adds default ports (80 for HTTP, 443 for HTTPS) when no port is specified.
+func GetProxyPorts(portProtocolMap map[factory.PortProtocol]bool) {
+	proxyEnvNames := sets.NewString("http_proxy", "https_proxy", "HTTP_PROXY", "HTTPS_PROXY")
+
+	// Get proxy environment variables and parse them for additional explicit ports
+	proxyEnvVars := utils.GetProxyEnvVars()
+
+	for _, envVar := range proxyEnvVars {
+		// Process proxy environment variables
+		if proxyEnvNames.Has(envVar.Name) {
+			if port := parsePortProtocolFromURL(envVar.Value); port != nil {
```

Comment:
corrected:  
In the case where the parseUrl fails, it will end up running url.Parse() twice .    
maybe the line above could be changed to something like:  
`if (envVar.Name == "http_proxy" || envVar.Name == "https_proxy") && envVar.Value != "" {`
in that case you wouldn't need to create the set and the extra nested if.   lmk if you want to chat.

Reply:
or you may consider creating a separate util method just for http env vars?

Reply:
I have a set here so that I can handle the different cases for the environment variable names.
I have consolidated this however so that `url.Parse` is not called twice.